app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,87 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[show edit update destroy]
+  # GET /recipes or /recipes.json
+  def index
+    # @recipes = Recipe.all
+    @recipes = current_user.recipes
+  end
+
+  # GET /recipes/1 or /recipes/1.json
+  def show
+    @recipe = Recipe.find(params[:id])
+    authorize! :read, @recipe
+  end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+    @recipe.steps.build # Build an empty step for the form
+    @user = current_user
+  end
+
+  # GET /recipes/1/edit
+  def edit; end
+
+  def create
+    @recipe = current_user.recipes.build(recipe_params)
+
+    if @recipe.save
+      redirect_to recipes_path, notice: 'Recipe was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  # PATCH/PUT /recipes/1 or /recipes/1.json
+  def update
+    respond_to do |format|
+      if @recipe.update(recipe_params)
+        format.html { redirect_to recipe_url(@recipe), notice: 'Recipe was successfully updated.' }
+        format.json { render :show, status: :ok, location: @recipe }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /recipes/1 or /recipes/1.json
+  def destroy
+    @recipe = Recipe.find(params[:id])
+    @recipe.recipe_foods.destroy_all
+    @recipe.destroy
+    redirect_to recipes_path, notice: 'Recipe eliminated successfully.'
```

Comment:
- [ ] Even if we display only the recipes owned by logged-in user, It would be nice to double check that only a recipe's owner can delete the recipe for security reasons.